internal/contentmanager/contentmanager.go

```diff
@@ -0,0 +1,203 @@
+// (unstructured.UnstructuredList for list kinds).
+//
+// If a provided client.Object does not set a Version or Kind field in its
+// GroupVersionKind, an error will be returned.
+func buildScheme(objs []client.Object) (*runtime.Scheme, error) {
+	scheme := runtime.NewScheme()
+	// The ClusterExtension types must be added to the scheme since its
+	// going to be used to establish watches that trigger reconciliation
+	// of the owning ClusterExtension
+	if err := v1alpha1.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("adding operator controller APIs to scheme: %w", err)
+	}
+
+	for _, obj := range objs {
+		gvk := obj.GetObjectKind().GroupVersionKind()
+		listKind := obj.GetObjectKind().GroupVersionKind().Kind + "List"
+
+		// If the Kind or Version is not set in an object's GroupVersionKind
+		// attempting to add it to the runtime.Scheme will result in a panic.
+		// To avoid panics, we are doing the validation and returning early
+		// with an error if any objects are provided with a missing Kind or Version
+		// field
+		if gvk.Kind == "" {
+			return nil, fmt.Errorf(
+				"adding %s to scheme; object Kind is not defined",
+				obj.GetName(),
+			)
+		}
+
+		if gvk.Version == "" {
+			return nil, fmt.Errorf(
+				"adding %s to scheme; object Version is not defined",
+				obj.GetName(),
+			)
+		}
+
+		if !scheme.Recognizes(gvk) {
+			// Since we can't have a mapping to every possible Go type in existence
+			// based on the GVK we need to use the unstructured types for mapping
+			u := &unstructured.Unstructured{}
+			u.SetGroupVersionKind(gvk)
+			ul := &unstructured.UnstructuredList{}
+			ul.SetGroupVersionKind(gvk.GroupVersion().WithKind(listKind))
+
+			scheme.AddKnownTypeWithName(gvk, u)
+			scheme.AddKnownTypeWithName(gvk.GroupVersion().WithKind(listKind), ul)
+			// Adding the common meta schemas to the scheme for the GroupVersion
+			// is necessary to ensure the scheme is aware of the different operations
+			// that can be performed against the resources in this GroupVersion
+			metav1.AddToGroupVersion(scheme, gvk.GroupVersion())
+		}
+	}
+
+	return scheme, nil
+}
+
+// ManageContent configures a controller-runtime cache.Cache and establishes watches for the provided resources.
+// It utilizes the provided ClusterExtension to set a DefaultLabelSelector on the cache.Cache
+// to ensure it is only caching and reacting to content that belongs to the ClusterExtension.
+// For each client.Object provided, a new source.Kind is created and used in a call to the Watch() method
+// of the provided controller.Controller to establish new watches for the managed resources.
+func (i *instance) ManageContent(ctx context.Context, ctrl controller.Controller, ce *v1alpha1.ClusterExtension, objs []client.Object) error {
+	// Return nil if the objs slice is empty before any further action
+	if len(objs) == 0 || ce == nil || ctrl == nil {
+		return nil
+	}
+
+	cfg, err := i.rcm(ctx, ce, i.baseCfg)
```

Comment:
Should this be the updated `ctx`?
i.e. the following from L182 should be here?
```
   ctx, cancel := context.WithCancel(ctx)

Reply:
I could do a new context if folks felt more comfortable with that, but in playing around with moving it I got a notification on my text editor about having to call the `cancel()` function every time we return. Seems like all code paths should use the cancel function or you risk a "context leak".

If we find that we need to move it later it should be a small lift

Reply:
`Watch()` is a non-blocking function with respect to the cache running, so it makes sense to use the context passed in from the caller everywhere _except_ for the cache's context that needs to be cancelled.

I do think the cache's context should be derived from `context.Background()` though. It's lifecycle is separate from the lifecycle of the passed in context (which is the context passed into `Reconcile`, which theoretically can be cancelled as soon as `Reconcile` returns). And we wouldn't want that canceling our cache's context.

Reply:
Sounds reasonable to me. Will update it to be derived from `context.Background()`

Reply:
Should be done in https://github.com/operator-framework/operator-controller/pull/1001/commits/0999fb28b4cfb39e0c3804b0ea4683119a8e285c